Settings.c

```diff
@@ -146,7 +147,6 @@ static bool Settings_read(Settings* this, const char* fileName, unsigned int ini
       }
       if (String_eq(option[0], "fields")) {
          readFields(this->fields, &(this->flags), option[1]);
-         didReadFields = true;
```

Comment:
Is the rework from `didReadFields` to `didReadAny` really needed?
The `fields` key defines the list of enabled process field columns, which every configuration file should contain.

Reply:
The reason for this patch is mostly that if the configuration file does not contain the `fields` key, the rest of the configuration keys are not applied either.

When manually editing the file and this can be confusing.

Reply:
To quote the source on this:
https://github.com/htop-dev/htop/blob/master/Settings.c#L327
> The parser is also very primitive, **and not human-friendly**.

Reply:
I still think the approach from @bjpbakker adds value as it cleans up the code and makes it more what you'd expect the relationship between compiled-in defaults and a setting file to be.
So ACK from DLange.

Reply:
Never really any objection from my side either. Just marked it as "Default change" due to this change of overall behavior.